lib/helpers/stack.ex

```diff
@@ -0,0 +1,23 @@
+defmodule Helpers.Stack do
+  @moduledoc """
+  This module provides functions to help mock user input for testing purposes.
+  Credit for this module goes to Lucas PenzeyMoog's blog post "Mock User Input for Integration Tests in Elixir/ExUnit" (https://medium.com/@lucaspenzeymoog/mock-user-input-for-integration-tests-in-elixir-exunit-a7f179fdfc53)
+  """
+
+  use Agent
```

Comment:
**quibble (non-blocking)** This is violating one of our constraints for Elixir TicTacToe. Do you know which one?

Reply:
Oh no! Is Agent a mocking library?

Reply:
😄 

It is not a mocking library, but it introduces statefulness.

Reply:
So it mutates state?

Reply:
Yeah

Reply:
If you recall a while back I mentioned how we could use "messages" to test a callback function was called. To do that we could use `send(self, {:some, :message})`, and `assert_received`.

`Agent` is a convenience module that encapsulates a common `send` and `receive` pattern using Elixir/Erlang "processes", which have a mechanism to hold and change state.

You might check out these docs on the Elixir site.
* [Processes](https://elixir-lang.org/getting-started/processes.html)
* [Agent](https://elixir-lang.org/getting-started/mix-otp/agent.html)